### ISSUE TITLE
where is the custom parser plugin 

### ISSUE THREAD

**Issue by shekarneo** [mrscarletzhou/retinaface-deepstream-python#5]
Hi, i am unable to find the parse-bbox-func-name=NvDsInferParseCustomRetinaface
custom-lib-path=/opt/models/retinaface/nvdsinfer_customparser/libnvdsinfer_custom_impl_retinaface.so

**Comment by zhouyuchong** [comment]
@shekarneo I've updated the repo. Now you can compile to make this custom post-proccess lib.

**Comment by shekarneo** [comment]
@zhouyuchong thank you. and if i convert pytorch model to ONNX will this parser works?.
i tried `https://github.com/wang-xinyu/tensorrtx/issues/1327` this to convert into engine file, but failed.

**Comment by zhouyuchong** [comment]
@shekarneo There are some differences. In a word, `tensorrtx` does finish decoding the raw output of model(trans anchor and NMS things). If you use your own ONNX file or pt file to generate tensorrt plan file(engine), you should modify the parser to decode its raw output.

**Comment by shekarneo** [comment]
@zhouyuchong thank you, it seems retinaface trt repo has some issues with ternsoRT 8.x version.

**Comment by zhouyuchong** [comment]
@shekarneo yeah, I've forked and fix them. You can try [this](https://github.com/zhouyuchong/tensorrtx). Hope it works.

**Comment by shekarneo** [comment]
i am able to create tensorrt engine by changing USE_INT8 to USE_FP32, but got these results.
![0_result](https://github.com/zhouyuchong/retinaface-deepstream-python/assets/25033310/fd002640-08a2-4b29-922e-6df26d47c802)

**Comment by shekarneo** [comment]
hi @zhouyuchong  i tried to run the retinaface with deepstream python test3 sample but got the below error, But i have the file in that path.

`ERROR: Could not open lib: /home/sensormatic/Neo/deepstream_python_apps/apps/deepstream-retinaface/models/nvdsinfer_customparser/libnvdsinfer_custom_impl_retinaface.so, error string: /home/sensormatic/Neo/deepstream_python_apps/apps/deepstream-retinaface/models/nvdsinfer_customparser/libnvdsinfer_custom_impl_retinaface.so: cannot open shared object file: No such file or directory
0:00:00.748377492 2556271      0x92c7830 ERROR                nvinfer gstnvinfer.cpp:640:gst_nvinfer_logger:<primary-inference> NvDsInferContext[UID 1]: Error in NvDsInferContextImpl::initialize() <nvdsinfer_context_impl.cpp:1257> [UID = 1]: Could not open custom lib: (null)
0:00:00.748463573 2556271      0x92c7830 WARN                 nvinfer gstnvinfer.cpp:846:gst_nvinfer_start:<primary-inference> error: Failed to create NvDsInferContext instance
0:00:00.748510006 2556271      0x92c7830 WARN                 nvinfer gstnvinfer.cpp:846:gst_nvinfer_start:<primary-inference> error: Config file path: configs/retinaface_pgie_config.txt, NvDsInfer Error: NVDSINFER_CUSTOM_LIB_FAILED
0:00:00.748648567 2556271      0x92c7830 WARN                GST_PADS gstpad.c:1142:gst_pad_set_active:<primary-inference:sink> Failed to activate pad
Error: gst-resource-error-quark: Failed to create NvDsInferContext instance (1): /dvs/git/dirty/git-master_linux/deepstream/sdk/src/gst-plugins/gst-nvinfer/gstnvinfer.cpp(846): gst_nvinfer_start (): /GstPipeline:pipeline0/GstNvInfer:primary-inference:
Config file path: configs/retinaface_pgie_config.txt, NvDsInfer Error: NVDSINFER_CUSTOM_LIB_FAILED
Exiting app`

**Comment by zhouyuchong** [comment]
@shekarneo 
```
/home/sensormatic/Neo/deepstream_python_apps/apps/deepstream-retinaface/models/nvdsinfer_customparser/libnvdsinfer_custom_impl_retinaface.so: cannot open shared object file: No such file or directory
```
plz check again. And BTW, this lib should be compiled again if you use it on Jetson platform since the original one was compiled on dgpu platform.

**Comment by shekarneo** [comment]
can you provide compile instructions here

**Comment by shekarneo** [comment]
@zhouyuchong  thans for pointing it, i have recompiled and it is working, but i am not able to Deserialize  the trt engine file

`ERROR: [TRT]: 1: [pluginV2Runner.cpp::load::292] Error Code 1: Serialization (Serialization assertion creator failed.Cannot deserialize plugin since corresponding IPluginCreator not found in Plugin Registry)
ERROR: [TRT]: 4: [runtime.cpp::deserializeCudaEngine::49] Error Code 4: Internal Error (Engine deserialization failed.)`

**Comment by zhouyuchong** [comment]
@shekarneo there is another trt decoder plugin when you finish generate `tensorrtx` plan file. You should pre-load it by [this](https://github.com/zhouyuchong/retinaface-deepstream-python/blob/028410e9572e3167d47dd1ba6f3d1da25377a5f6/main.py#L21) or using command 
```
LD_PRRLOAD={PATH}
```

**Comment by shekarneo** [comment]
@zhouyuchong  thanks for your support i am able to run the model successfully.

There is accuracy  drop in the model i am using FP32, but still it is missing the detections and giving lots of false positives. Did you face the same?

**Comment by zhouyuchong** [comment]
@shekarneo  Nope. maybe you should check your parser-lib. There are little difference if you didn't set the right config properties. But it won't give you false positives.

**Comment by shekarneo** [comment]
@zhouyuchong  I am currently using 1920X1080 input size, do i need to change it ti input-height:480
input-width:640

**Comment by zhouyuchong** [comment]
@shekarneo You can have a try. The reason could be any details. I can't help you without little info.

**Comment by shekarneo** [comment]
@zhouyuchong i tried but it did not worked, even with your code it did not gave better results. I think the problem is with the engine generation

and may i know what info you need

**Comment by zhouyuchong** [comment]
@shekarneo I'll check these below if I were you:
1. test the inference plan file gened by `tensorrtx` alone. `tensorrtx` repo have provided script to test inference on images.
2. If step 1 is ok, I'll use a single pic to make a video. Then output or record raw-output of inference in deepstream. The output should be almost the same between step 1 and in deepstream.
3. If they are distinct, I'll compare the results and check codes. Mostly you'll find some rules.

**Comment by shekarneo** [comment]
@zhouyuchong 
I tested with this `sudo ./retina_r50 -d  // deserialize model file and run inference.` but there also accuracy was not good

**Comment by zhouyuchong** [comment]
@shekarneo What do you mean accuracy was not good?

**Comment by shekarneo** [comment]
its giving false positive bboxes and some time not detecting the faces

**Comment by zhouyuchong** [comment]
@shekarneo That's weird :thinking: then you should back to step 1 to check `tensorrtx` codes.
